vehicle/src/Vehicle/Prelude/IO.hs

```diff
@@ -97,6 +100,23 @@ getVehiclePath = do
   liftIO $ createDirectoryIfMissing False vehiclePath
   return vehiclePath
 
+-- | Writes a file atomically by first acquiring a lock on it.
+safeWriteToFile :: (Handle -> a -> IO ()) -> FilePath -> a -> IO ()
+safeWriteToFile writeOp file contents = do
+  handle <- openFile file WriteMode
+  hLock handle ExclusiveLock
+  writeOp handle contents
+  hClose handle
+
+-- | Writes a file atomically by first acquiring a lock on it.
+safeReadFromFile :: (Handle -> IO a) -> FilePath -> IO a
+safeReadFromFile readOp file = do
+  handle <- openFile file ReadMode
+  hLock handle SharedLock
+  result <- readOp handle
```

Comment:
Okay @wenkokke, I'm stumped here. This is erroring because it's lazily starting to read the file after I've called `hclose handle`. So how do I make the reading of the file strict? 

I've tried `!result <- readOp handle` to no avail...